### ISSUE TITLE
MacOS BigSur moves into full screen desktop too early, missing the screenshot

### ISSUE THREAD

**Issue by heyheyhello** [flameshot-org/flameshot#1650]
**Flameshot version**

0.9.0 and checking for updates says I'm on latest

**Describe the bug**

I started experiencing this with the BigSur update here's a video:

https://user-images.githubusercontent.com/44614862/120225209-ab4e8780-c1f9-11eb-938c-c26049fe7125.mov

**To Reproduce**

Press the hotkey Cmd+Option+Shift+4 or "Take Screenshot" in the menu to start Flameshot.

**Expected behavior**

I expect it to be able to grab the screen before moving into the full screen mode, but it misses it.

**System Information**

MacOS 11.3.1

**Comment by heyheyhello** [comment]
Related to #1636? But was closed without explanation or fix...

**Comment by heyheyhello** [comment]
A potential easy workaround would be to provide a configuration option like

- [x] Snapshot screen and move into a new macOS workspace

Which could be **disabled** since not everyone is taking screenshots of time-sensitive things (although it is very useful for that). I'm just taking a screenshot of my editor and I know the content isn't moving anywhere, so being able to draw/annotate with Flameshot _in the same desktop window_ would be fine and would avoid this BigSur bug

**Comment by borgmanJeremy** [comment]
Are you sure you have given flameshot permission to screen record?

**Comment by heyheyhello** [comment]
Yes 
<img width="780" alt="image" src="https://user-images.githubusercontent.com/44614862/120230452-2cab1780-c204-11eb-821c-e8399e775b02.png">

**Comment by heyheyhello** [comment]
Oh, you're right - it's a macOS bug not a Flameshot bug. The hint was with this dialog opening on Flameshot restart 
<img width="529" alt="image" src="https://user-images.githubusercontent.com/44614862/120230712-c83c8800-c204-11eb-9ee8-8679ade2920f.png">

Even though clicking "Open System Preferences" shows that Flameshot is granted permission.

I had to fully _remove_ (using the "-" button) the application and then let it re-add itself. If I only did an uncheck/recheck it still didn't work. Thanks!
